pkg/services/hcloud/loadbalancer/loadbalancer.go

```diff
@@ -138,163 +133,152 @@ func (s *Service) reconcileNetworkAttachement(ctx context.Context, lb *hcloud.Lo
 
 func (s *Service) reconcileLBProperties(ctx context.Context, lb *hcloud.LoadBalancer) error {
 	var multierr []error
-	// Check if type has been updated
-	if s.scope.HetznerCluster.Spec.ControlPlaneLoadBalancer.Type != lb.LoadBalancerType.Name {
-		if _, err := s.scope.HCloudClient.ChangeLoadBalancerType(ctx, lb, hcloud.LoadBalancerChangeTypeOpts{
-			LoadBalancerType: &hcloud.LoadBalancerType{
-				Name: s.scope.HetznerCluster.Spec.ControlPlaneLoadBalancer.Type,
-			},
-		}); err != nil {
+	lbSpec := s.scope.HetznerCluster.Spec.ControlPlaneLoadBalancer
+
+	// check if type has been updated
+	if lbSpec.Type != lb.LoadBalancerType.Name {
+		opts := hcloud.LoadBalancerChangeTypeOpts{LoadBalancerType: &hcloud.LoadBalancerType{Name: lbSpec.Type}}
+		if _, err := s.scope.HCloudClient.ChangeLoadBalancerType(ctx, lb, opts); err != nil {
 			hcloudutil.HandleRateLimitExceeded(s.scope.HetznerCluster, err, "ChangeLoadBalancerType")
 			multierr = append(multierr, fmt.Errorf("failed to change load balancer type: %w", err))
```

Comment:
about `multierr`: Since caph uses Go 1.20, you could use errors.Join(): https://pkg.go.dev/errors#Join